### PR TITLE
refactor(ci): to use github app auth over long-lived credentials

### PR DESCRIPTION
**Pull request by tdeekens**
#### Summary

This refactors the authentication for the GitHub PAT for releasing via Changesets from a long-lived static credential to a short-lived dynamic credential issues through a GitHub App.

More information can be found in internal documentation or repositories.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -10,14 +10,22 @@ jobs:
     runs-on: ubuntu-latest
 
     steps:
+      # Get GitHub token via the CT Changesets App
+      - name: Generate GitHub token (via CT Changesets App)
+        id: generate_github_token
+        uses: tibdex/github-app-token@v2.1.0
+        with:
+          app_id: ${{ secrets.CT_CHANGESETS_APP_ID }}
+          private_key: ${{ secrets.CT_CHANGESETS_APP_PEM }}
```
Comment (CarlosCortizasCT) [inline .github/workflows/release.yml:19]: Are these secrets now configured at Github repository level?

Comment (tdeekens) [reply .github/workflows/release.yml:19]: They are configured on the organisation level actually like other apps. With the addition that they're also restricted to which (six in total) repositories they are exposed. A quite nice feature in GitHub :)

**Review by CarlosCortizasCT — APPROVED** [review]
Many thanks 👍

**Comment by emmenko** [comment]
Thanks @tdeekens !

Did you also remove the token from the repository secrets?

**Comment by tdeekens** [comment]
> Thanks @tdeekens !
> Did you also remove the token from the repository secrets?

I am migrating this across six repos. I have reminder in my calendar to remove all PATs across all those repos next Monday.

**Pull request merged**